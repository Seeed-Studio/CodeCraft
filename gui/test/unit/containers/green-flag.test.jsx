import React from 'react';
import {shallow} from 'enzyme';
import GreenFlag from '../../../src/containers/green-flag';
import renderer from 'react-test-renderer';
import VM from '../../../../vm/';

describe('GreenFlag Container', () => {
    let vm;
    beforeEach(() => {
        vm = new VM();
    });

    test('renders active state', () => {
        const component = renderer.create(
            <GreenFlag
                active
                vm={vm}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('renders inactive state', () => {
        const component = renderer.create(
            <GreenFlag
                active={false}
                vm={vm}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('triggers onClick when active', () => {
        const onClick = jest.fn();
        const componentShallowWrapper = shallow(
            <GreenFlag
                active
                vm={vm}
                onClick={onClick}
            />
        );
        componentShallowWrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });

    // @todo: Test for handles key events.
    // @todo: Test project run start.
    // @todo: Test project run stop.
});
