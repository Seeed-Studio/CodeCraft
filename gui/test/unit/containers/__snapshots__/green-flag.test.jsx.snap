// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenFlag Container renders active state 1`] = `
<img
  className="undefined"
  draggable={false}
  onClick={[Function]}
  src="test-file-stub"
  title="Go"
/>
`;

exports[`GreenFlag Container renders inactive state 1`] = `
<img
  className=""
  draggable={false}
  onClick={[Function]}
  src="test-file-stub"
  title="Go"
/>
`;
